src/Core/Vault/Services/Implementations/CipherService.cs

```diff
@@ -648,19 +648,23 @@ public async Task DeleteFolderAsync(Folder folder)
             }
         }
 
-        var existingFolders = new List<Folder>();
-        // Init. ids for folders
+        var foldersGuids = folders.Select(f => f.Id).ToList();
+
+        var dbfolders = (await _folderRepository.GetManyByManyIdsAndUserIdAsync(foldersGuids, ciphers[0].UserId ?? Guid.Empty)).ToList();
```

Comment:
Why not just grab all folders in the list, filter out the ones with the wrong user id, and that's already our existing ciphers list?

Reply:
Second nit here: shouldn't this limit you to importing into the current users' vault? not whatever user is specified in the import

Reply:
We need to keep order to create the relationship. We are comparing CipherList position with FolderList position so because of this I can't just get a list of new ones right here.

Also we need the existing ones to ensure the name is updated from the import, the use case should also update the existing folders names, by this reason I need to update them aswell.

Reply:
The userId is set to the current user on the Controller